tests/test_stdiomgr_base.py

```diff
@@ -26,13 +26,64 @@
 
 """
 
+import collections.abc
 import io
 import sys
 import warnings
 
+try:
+    from contextlib import AbstractContextManager
```

Comment:
this could be `from stdio_mgr import AbstractContextManager` , but I slightly prefer it like this, so it is clear that this is a bit of strange voodoo.

Should include a note that the else is for Python <3.6

Reply:
&lt;nod&gt;, I definitely prefer the current explicit try/except.

Definitely favor adding the <3.6 note.

What benefit is gained from including `AbstractContextManager` in the inheritance tree? The code doesn't seem to make explicit use of it.

Reply:
No real benefit. This is just keeping the base, previously obtained via _GeneratorContextManager via @contextmanager

Reply:
Indeed -- might as well keep it, on the off-chance someone has reason to test the inheritance.

Reply:
The benefit is it is *the* registered ABC for context managers, despite the abstractness.  And also `typing.ContextManager` exists for it, in case anyone wants to enforce type safety.

Implementing `typing` would be a good idea, and I've found the backports to py27 & py34 are very solid, but using it with py27 is IMO messy and not worth the trouble.

Reply:
I haven't touched typing at all in my projects yet, so I don't have any hands-on sense of the value it brings.  (Also, I tend to develop lately in simple editors without intellisense/code-completion, so I personally wouldn't benefit there, for now.)

To enhance my perspective on it, what sorts of practical benefits would you anticipate from adding it to this project?  Mostly for users, when they're developing with stdio-mgr? For us, as development progresses?

Reply:
The main use of `typing` I can think of is users passing the CM object around, with signatures that check for data types.


More likely people will be doing that in 2024.